tests/test_config_parsing/config_dict_generator.py

```diff
@@ -310,15 +367,79 @@ def config_dicts(draw):
     return config_dict
 
 
-def to_config_file(filename, config_dict):
-    with open(filename, "w+") as config:
+def to_config_file(filename, config_dict):  # pylint: disable=too-many-branches
```

Comment:
This function seems to perhaps have outlived being just one if/elif chain. My solution here would be to use a separate structure which we turn into this ConfigKeys. This we could keep in the tests for now but eventually turn into a intermediate representation for the parser.

What I am thinking is to have a 

```python
@dataclass
def Field:
   name: str = ""
   var_type: str = ""
   ...
   
   def to_file_contents(self) -> str:
     return " ".join ( ["FIELD", ...]
```

so that `to_config_file` becomes

```python
 to_config_file(file_name, config_ast):
     with open(file_name) as f:
         for keyword in config_ast:
             f.write(keyword.to_file_contents())
 ```

Reply:
Thanks.
- Removed WIP from commit message
- Squashed mentioned commits
- Split lint commit and fixup's where applicable